app/backend/src/clj/teet/file/file_commands.clj

```diff
@@ -81,18 +83,30 @@
     {:url (file-storage/upload-url key)
      :file (d/pull (:db-after res) '[*] file-id)}))
 
+(defn maybe-vektorio-delete! [db file-eid]
+  (let [vektorio-enabled? (environment/feature-enabled? :vektorio)
+        vektorio-config (environment/config-value :vektorio)
+        project-eid (project-db/file-project-id db file-eid)]
+    (log/debug "delete corresponding model from vektorio? " (some? vektorio-enabled?))
+    (when vektorio-enabled?
+      (vektorio/delete-file-from-project! db vektorio-config project-eid file-eid))))
 
 (defcommand :file/delete-attachment
   {:doc "Delete an attachment"
    :context {:keys [user db]}
    :payload {:keys [file-id attached-to]}
    :project-id nil
    :authorization {}
+   ;; this command has modal behaviour depending on whether attached-to tuple is supplied, or nil.
+   ;; if it is supplied, permission check and delete-attachment call go to the multimethod that
+   ;; will do a permission check depending on the attachment type.
    :pre [(or attached-to
              (file-db/own-file? db user file-id))]
-   :transact (if attached-to
-               (file-db/delete-attachment db user file-id attached-to)
-               [(deletion-tx user file-id)])})
+   :transact (let [file-delete-tx (if attached-to
+                                    (file-db/delete-attachment db user file-id attached-to)
+                                    [(deletion-tx user file-id)])]
+               (maybe-vektorio-delete! db file-id)
```

Comment:
can attachments be models? or should they only be proper task files

Reply:
Good question. The delete code can check for existence of the vektorio attributes. I actually had a bug there since I hadn't tested with files that don't have vektorio models, i fixed that now in vektorio-core/delete-file-from-project! fn